src/test/java/org/fungover/mmotodo/task/TaskControllerTest.java

```diff
@@ -0,0 +1,246 @@
+
+        // language=GraphQL
+        String query = """
+                    mutation {
+                         addTask(task: {
+                            title: "new task",
+                            description: "this is the new task",
+                            status: "started"
+                        }) {
+                            title
+                            description
+                            status
+                        }
+                    }
+                """;
+
+        String expected = objectMapper.writeValueAsString(taskDto);
+
+        graphQlTester.document(query)
+                .execute()
+                .path("addTask")
+                .matchesJson(expected);
+    }
+
+    @Test
+    void ShouldHandleSuccessDeleteTaskResponse() throws Exception {
+        Mockito.when(taskService.deleteTask(1)).thenReturn(true);
+
+        // language=GraphQL
+        String query = "mutation { deleteTask(id: 1) }";
+
+        String expected = objectMapper.writeValueAsString("Task with id: 1 is deleted");
+
+        graphQlTester.document(query)
+                .execute()
+                .path("deleteTask")
+                .matchesJson(expected);
+    }
+
+    @Test
+    void ShouldHandleFailedDeleteTaskResponse() throws Exception {
+        Mockito.when(taskService.deleteTask(1)).thenReturn(false);
+
+        // language=GraphQL
+        String query = "mutation { deleteTask(id: 1) }";
+
+        String expected = objectMapper.writeValueAsString("Could not delete task with id 1");
+
+        graphQlTester.document(query)
+                .execute()
+                .path("deleteTask")
+                .matchesJson(expected);
+    }
+
+    @Test
+    void ShouldUpdateTaskTitle() throws Exception {
+        TaskUpdateDto updateDto = new TaskUpdateDto(1, "Updated Task", null, null, null, null);
+        Task task = Task.createTestTask(1);
+        task.setTitle(updateDto.title());
+        task.setDescription(updateDto.description());
+        task.setStatus(updateDto.status());
+
+        Mockito.when(taskService.updateTask(updateDto)).thenReturn(task);
+
+        // language=GraphQL
+        String query = "mutation { updateTask(task: { id: 1, title: \"Updated Task\"}) { title }}";
+
+        String expected = objectMapper.writeValueAsString(new Object() {
+            public final String title = "Updated Task";
+        });
+
+        graphQlTester.document(query)
+                .execute()
+                .path("updateTask")
+                .matchesJson(expected);
+
+    }
+
+    private Task createTask(String title) {
+        Task task = Task.createTestTask(counter);
+        task.setTitle(title);
+        task.setDescription("Test task");
+        Category category = new Category();
+        category.setId(counter);
+        task.setCategory(category);
+        Tag tag = new Tag();
+        tag.setId(counter);
+        task.setTag(tag);
+        counter++;
+        return task;
+    }
+}
```

Comment:
Really nice tests here, well structured, you cover a lot of scenarios and the code is clean.